libraries/ms-common/inc/bootloader_can.h

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+// Bootloader Design Confluence: https://uwmidsun.atlassian.net/l/c/76h1GfbH
+// This module allows the sending and receiving of CAN messages in the bootloader's CAN ID format.
+// It allows registering a callback which will be called when a bootloader CAN message is received
+// with client controller board ID 0 (msg_id == 0).
+// Requires interrupts, gpio, the event queue, soft timers and CAN to be initialized.
+// If only bootloader datagrams are to be transmitted, the bootloader module needs to be initialized
+// otherwise, initializing CAN is sufficient
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "can.h"
+#include "can_msg_defs.h"
+#include "can_transmit.h"
+#include "status.h"
+
+// Called whenever a bootloader message is received with client controller board ID 0
+typedef StatusCode (*BootloaderCallback)(const CanMessage *msg, void *context);

```

Comment:
Nitpick: could you call this `BootloaderCanCallback` instead? `BootloaderCallback` is a bit non-specific, we'll likely define more callbacks in the bootloader proper.